src/opm/input/eclipse/Schedule/Well/WellConnections.cpp

```diff
@@ -441,6 +508,228 @@ namespace Opm {
+            rw = 0.5*unit::feet;
+
+        // Get the grid 
+        auto ecl_grid = grid.get_grid();
+
+        // Calulate the x,y,z coordinates of the begin and end of a perforation
+        std::vector<double> coord_top{0,0,0};
+        std::vector<double> coord_bot{0,0,0};
+        for (size_t i = 0; i < 3 ; ++i) {
+            coord_top[i] =  Opm::linearInterpolation(this->md, this->coord[i], perf_top.getSIDouble(0));
+            coord_bot[i] =  Opm::linearInterpolation(this->md, this->coord[i], perf_bot.getSIDouble(0));
+        }
+        external::cvf::Vec3d p_top(coord_top[0], coord_top[1], coord_top[2]);
+        external::cvf::Vec3d p_bot(coord_bot[0], coord_bot[1], coord_bot[2]);
+        std::vector<external::cvf::Vec3d> points{p_top, p_bot};
+        std::vector<double> md_interval{perf_top.getSIDouble(0), perf_bot.getSIDouble(0)};
+        
+        external::cvf::ref<external::RigWellPath> wellPathGeometry = new external::RigWellPath;
+        wellPathGeometry->setWellPathPoints(points);
+        wellPathGeometry->setMeasuredDepths(md_interval);
+        external::cvf::ref<external::RigEclipseWellLogExtractor> e = new external::RigEclipseWellLogExtractor(wellPathGeometry.p(), *ecl_grid, cellSearchTree);
+        
+        // Keep the AABB search tree of the grid  to avoid redoing an expensive calulation 
+        cellSearchTree = e->getCellSearchTree();
+
+        // This gives the intersected grid cells IJK, cell face entrance & exit cell face point and connection length 
+        auto intersections = e->cellIntersectionInfosAlongWellPath();
+
+        int I{0};
+        int J{0};
+        int k{0};
+        for (size_t is = 0; is < intersections.size(); ++is){
+            auto ijk = std::array<int, 3>{};
+            ijk = ecl_grid->getIJK(intersections[is].globCellIndex);
+            I = ijk[0];
+            J = ijk[1];
+            k = ijk[2];
+            // std::cout<< "I: " << I << " J: " << J << " K: " << k << std::endl;
+            external::cvf::Vec3d connection_vector = intersections[is].intersectionLengthsInCellCS;
+
+
+            const CompletedCells::Cell& cell = grid.get_cell(I, J, k);
+
+           if (!cell.is_active()) {
+                auto msg = fmt::format("Problem with COMPTRAJ keyword\n"
+                                       "In {} line {}\n"
+                                       "The cell ({},{},{}) in well {} is not active and the connection will be ignored", location.filename, location.lineno, I,J,k, wname);
+                OpmLog::warning(msg);
+                continue;
+            }
+            const auto& props = cell.props;
+            double CF = -1;
+            double Kh = -1;
+            double r0 = -1;
+            auto ctf_kind = ::Opm::Connection::CTFKind::DeckValue;
+
+            if (defaultSatTable)
+                satTableId = props->satnum;
+
+            auto same_ijk = [&]( const Connection& c ) {
```

Comment:
Maybe just capture what you need? Even by value should be fine `... = [I, J, K](...)`

Reply:
This inline function returns a boolean  and is used later in `std::find_if`. This construction is also there in `loadCOMPDAT`.

Reply:
I think there might be a misunderstanding. I was talking about `[&]`, which captures all local variable by reference to make them usable inside your lambda. As you are just using I, j, and K, I was suggesting to only capturing them.

Anyway, this is probably not a big deal, and just me justifying my existence as a reviewer :sweat_smile:. Hence feel free to leave as is.

Reply:
Ah I see, indeed I did misunderstood. I followed your suggestion.